### ISSUE TITLE
Training Recipe

### ISSUE THREAD

**Issue by WenliangGuo** [CaptainCook4D/error_recognition#3]
Thank you for presenting this awesome work!

When I tried to reproduce the error recognition model, I found that the training parameters in the code and those mentioned in the paper didn't match. For example, the paper says for the step-level model, the learning rate is 1e-3, the batch size is 512, the loss function is Binary Cross Entropy Loss (BCE Loss) with a weight of 1.5 for the positive classes, and the training epoch is set to 50. However, when I look at the code, the batch size is 1 using a standard BCD loss w/o weighting for the positive class, and train for 10 epochs. The similar thing happens on the recording-level model as well. 

I am also wondering about the differences between the  _sub-step_ and _step metric_ printed in the training log. Which one is used as the final result for the error recognition model?

I would appreciate it if you could update the training configuration and address my confusion so I can reproduce the results reported in the paper and better follow your work. Thank you so much.

**Comment by rohithpeddi** [comment]
Hi @WenliangGuo ,

Sorry for the late reply. 

------

> Configuration

Our reported results follow exact hyperparameter configuration in the paper. 
Please see [LINE](https://github.com/CaptainCook4D/error_recognition/blob/35ef284a673230a3df09dd971fc5163414f23b2b/base.py#L179)
The code is for a reference of how we trained and evaluated our models. 
Please use the configuration settings provided in the paper. 
The cause for confusion in settings might be due to several ablation studies we did inorder to determine best performing results. 
We have also released all the trained models which you can directly use for evaluation. 
Please look at [ISSUE](https://github.com/CaptainCook4D/error_recognition/issues/1#issuecomment-2558346239)

-------

> Sub-step vs step

**Sub-step** - Correspond to predictions for 1 second segments of steps.
**Step** - Correspond to predictions for the complete step. 

Please see [LINE](https://github.com/CaptainCook4D/error_recognition/blob/35ef284a673230a3df09dd971fc5163414f23b2b/base.py#L407) for exact method employed to calculate the Step/Substep metrics. 


I will also update this in the README file.
